backend-tests/tests/test_deployments.py

```diff
@@ -1446,3 +1450,589 @@ class TestDeploymentsStatusUpdateEnterprise(TestDeploymentsStatusUpdateBase):
     def test_deployment_status_update(self, clean_mongo):
         _user, _tenant, user_token, devs = setup_devices_and_management_mt(5)
         self.do_test_deployment_status_update(clean_mongo, user_token, devs)
+
+
+def create_tenant(name, username, plan):
+    tenant = create_org(name, username, "correcthorse", plan=plan)
+    user = tenant.users[0]
+    r = ApiClient(useradm.URL_MGMT).call(
+        "POST", useradm.URL_LOGIN, auth=(user.name, user.pwd)
+    )
+    assert r.status_code == 200
+    user.utoken = r.text
+
+    return tenant
+
+
+@pytest.fixture(scope="function")
+def setup_tenant(clean_mongo):
+    return create_tenant("foo", "foo@tenant.com", "enterprise")
+
+
+def predicate(attr, scope, t, val):
+    return {"attribute": attr, "scope": scope, "type": t, "value": val}
+
+
+def create_filter(name, predicates, utoken):
+    f = {"name": name, "terms": predicates}
+
+    r = (
+        ApiClient(inventory_v2.URL_MGMT)
+        .with_auth(utoken)
+        .call("POST", inventory_v2.URL_FILTERS, f)
+    )
+    assert r.status_code == 201
+
+    f["id"] = r.headers["Location"].split("/")[1]
+    return f
+
+
+def create_dynamic_deployment(
+    name, predicates, utoken, max_devices=None, phases=None, status_code=201
+):
+    f = create_filter(name, predicates, utoken)
+
+    api_dep_v2 = ApiClient(deployments_v2.URL_MGMT)
+    api_dep_v1 = ApiClient(deployments.URL_MGMT)
+
+    depid = None
+    with get_mender_artifact(name) as filename:
+
+        upload_image(filename, utoken)
+
+        deployment_req = {
+            "name": name,
+            "artifact_name": name,
+            "filter_id": f["id"],
+        }
+
+        if max_devices is not None:
+            deployment_req["max_devices"] = max_devices
+
+        if phases is not None:
+            deployment_req["phases"] = phases
+
+        res = api_dep_v2.with_auth(utoken).call(
+            "POST", deployments_v2.URL_DEPLOYMENTS, deployment_req
+        )
+
+        assert res.status_code == status_code
+        if status_code != 201:
+            return None
+
+        depid = res.headers["Location"].split("/")[5]
```

Comment:
I would consider checking here fi indeed the Location matches what we expect, and get explicit error if not. As another test for the correct Location:.

Reply:
well, if the location is not what we assume - it will blow up the test and pretty quickly tell us where it failed, so I don't think it's necessary